### DISCUSSION TITLE
Running SSH-MITM on Android devices

### DISCUSSION THREAD

**Discussion by xkjdox** [ssh-mitm/ssh-mitm#83] [Q&A] [answered]

**Comment by manfred-kaiser** [comment]
What operating system are you using? Can you provide more information please?
From the screenshot, it looks, that you are try to install ssh-mitm on a mobile device. 

Architecture: X86_64, ARM, ARM64, ...
Operating System: Windows, Ubuntu, Debian, RedHat, ...
Python Version?
Complete Error Message as Text (no Screenshots)

SSH-MITM uses the python cryptography package, but this package is not distributed with my application, except for the AppImage and the Snap.
If the cryptography package is missing, pip tries to install it from pypi.org. But you can also install it with the package manager from you distribution (if provided).

As you wrote, the installation of the cryptography package failed. If you have problems during installing this package, you should read https://cryptography.io/en/latest/installation/

You should also check if your platform is supported by the python cryptography package.

You can try following steps:

1. Update pip to the latest version. This could solve problems during installation.
2. Check if you have a rust compiler installed. If your platform is not officially supported by the cryptography package, you have to provide a rust compiler
2. Perhaps some python development packages are missing - install the python development packages provided by your distribution
3. Perhaps your distribution provides a precompiled python cryptography package, which can be used.

**Comment by xkjdox** [comment]
here

Architecture: aarch64
Operating System: Termux
python version : Python 3.10.0
pip : 21.3.1
setuptools : 58.3.0
wheel : 0.37.0
setuptools-rust : 0.12.1

**Comment by manfred-kaiser** [comment, marked as answer]
**Run SSH-MITM on Android devices:**

SSH-MITM works with  https://play.google.com/store/apps/details?id=ru.iiec.pydroid3&hl=de&gl=US (Note: the mentioned app is not open source :disappointed:)

1. Install ssh-mitm with pip. You can find "Pip" in the menu on the left side
2. Open the terminal from the menu on the left side
3. Start ssh-mitm from the terminal. You can also use command line arguments

![Screenshot_20211025-144550](https://user-images.githubusercontent.com/37737811/138698029-fb473205-a042-425a-ace2-be570966aaa8.png)


**Note on Termux:**

I was not able to install the required packages in termux. The installation failed when I try to install "cffi", which is required by "cryptography".

**Reply by ChillVibesMushroom** [reply]
> **Run SSH-MITM on Android devices:**
> 
> SSH-MITM works with https://play.google.com/store/apps/details?id=ru.iiec.pydroid3&hl=de&gl=US (Note: the mentioned app is not open source 😞)
> 
> 1. Install ssh-mitm with pip. You can find "Pip" in the menu on the left side
> 2. Open the terminal from the menu on the left side
> 3. Start ssh-mitm from the terminal. You can also use command line arguments
> 
> ![Screenshot_20211025-144550](https://user-images.githubusercontent.com/37737811/138698029-fb473205-a042-425a-ace2-be570966aaa8.png)
> 
> **Note on Termux:**
> 
> I was not able to install the required packages in termux. The installation failed when I try to install "cffi", which is required by "cryptography".

I noticed that we can exploit laptops and towers but when it comes to Termux I can't exploit it to the full extent of ssh-mitm power I can only get the password hahaha I want to be able to mirror the shell are you having the same issue has this been asked before if so can you direct me to the right post

**Comment by ChillVibesMushroom** [comment]
When attempting to log into the ssh server it keeps saying my password is incorrect when I know that it isn't

**Reply by manfred-kaiser** [reply]
can you describe your scenario? please provide the command line options, which are used in this scenario.
